src/wallet/sqlite.cpp

```diff
@@ -612,7 +617,7 @@ std::unique_ptr<DatabaseCursor> SQLiteBatch::GetNewPrefixCursor(Span<const std::
 
 bool SQLiteBatch::TxnBegin()
 {
-    if (!m_database.m_db || sqlite3_get_autocommit(m_database.m_db) == 0) return false;
+    if (!m_database.m_db || m_database.HasActiveTxn()) return false;
```

Comment:
In 83c807a380b41787214806c6440236ea1a9dabd7 "sqlite: introduce HasActiveTxn method"

`HasActiveTxn()` checks for `m_db` already, so no need to do check that here too.

Same for `TxnCommit()` and `TxnAbort()`.

Reply:
upps done